### ISSUE TITLE
Sundown (and sunup) time isn't changing.

### ISSUE THREAD

**Issue by AdeenDragon** [multimokia/MAS-Submod-Auto-Atmos-Change#5]
The Sun Times aren't changing, and when I turn on and off "Auto Sun Times" in the Submods menu, I get the following error.

[code]
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "renpy/common/00gamemenu.rpy", line 173, in script
    $ ui.interact()
  File "renpy/common/00gamemenu.rpy", line 173, in <module>
    $ ui.interact()
  File "renpy/common/00action_other.rpy", line 484, in __call__
    rv = self.callable(*self.args, **self.kwargs)
  File "game/Submods/Auto Weather Change/Utilities/awc_utils.rpy", line 75, in toggleAST
    store.persistent._mas_sunrise = store.awc_dtToMASTime(store.awc_getSunriseDT())
  File "game/Submods/Auto Weather Change/Utilities/awc_utils.rpy", line 298, in awc_getSunriseDT
    awc_getObservation().get_weather().get_sunrise_time('date') - datetime.timedelta(seconds=time.timezone)
AttributeError: 'NoneType' object has no attribute 'get_weather'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "renpy/common/00gamemenu.rpy", line 173, in script
    $ ui.interact()
  File "C:\Users\morga\Desktop\Monika\renpy\ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\Users\morga\Desktop\Monika\renpy\python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "renpy/common/00gamemenu.rpy", line 173, in <module>
    $ ui.interact()
  File "C:\Users\morga\Desktop\Monika\renpy\ui.py", line 285, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "C:\Users\morga\Desktop\Monika\renpy\display\core.py", line 2526, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, **kwargs)
  File "C:\Users\morga\Desktop\Monika\renpy\display\core.py", line 3204, in interact_core
    rv = root_widget.event(ev, x, y, 0)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\transition.py", line 45, in event
    return self.new_widget.event(ev, x, y, st)  # E1101
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\screen.py", line 651, in event
    rv = self.child.event(ev, x, y, st)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 232, in event
    rv = d.event(ev, x - xo, y - yo, st)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 232, in event
    rv = d.event(ev, x - xo, y - yo, st)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 232, in event
    rv = d.event(ev, x - xo, y - yo, st)
  File "C:\Users\morga\Desktop\Monika\renpy\display\viewport.py", line 303, in event
    rv = super(Viewport, self).event(ev, x, y, st)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 232, in event
    rv = d.event(ev, x - xo, y - yo, st)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\layout.py", line 960, in event
    rv = i.event(ev, x - xo, y - yo, cst)
  File "C:\Users\morga\Desktop\Monika\renpy\display\behavior.py", line 889, in event
    return handle_click(self.clicked)
  File "C:\Users\morga\Desktop\Monika\renpy\display\behavior.py", line 824, in handle_click
    rv = run(action)
  File "C:\Users\morga\Desktop\Monika\renpy\display\behavior.py", line 306, in run
    return action(*args, **kwargs)
  File "renpy/common/00action_other.rpy", line 484, in __call__
    rv = self.callable(*self.args, **self.kwargs)
  File "game/Submods/Auto Weather Change/Utilities/awc_utils.rpy", line 75, in toggleAST
    store.persistent._mas_sunrise = store.awc_dtToMASTime(store.awc_getSunriseDT())
  File "game/Submods/Auto Weather Change/Utilities/awc_utils.rpy", line 298, in awc_getSunriseDT
    awc_getObservation().get_weather().get_sunrise_time('date') - datetime.timedelta(seconds=time.timezone)
AttributeError: 'NoneType' object has no attribute 'get_weather'

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Monika After Story 0.11.6
[/code]

I'm not certain of when it happened, but going by a calendar of sun down times, it looks like the last day it was accurate was ~ October 4th.  I'm using the submod auto updater, so I should be using the latest version of Auto Atmos Change.

The weather changes normally, when the forecast calls for rain, it's raining in MAS.

**Comment by multimokia** [comment]
Hmmm, still trying to figure this out. Doesn't seem to be any reason the time gets should fail given weather gets use the exact same call

**Comment by AdeenDragon** [comment]
Are there further information/tests that would be helpful?

**Comment by multimokia** [comment]
Could you by chance send your persistent over? You'll have to put it in a zip file for github to accept it. Just want to see if it's something on the api side

**Comment by AdeenDragon** [comment]
[persistent.zip](https://github.com/multimokia/MAS-Submod-Auto-Atmos-Change/files/5598182/persistent.zip)

Here you go, is there anything else that would help?

**Comment by multimokia** [comment]
Apologies for the late response.

Okay so this is interesting, has Monika ever asked you for your location at all? If so it's odd because the location isn't stored at all within the persistent. This much would explain at least partially why it's crashing. But not at all why the weather itself seems to function. I'm almost wondering if it's by sheer coincidence?

**Comment by AdeenDragon** [comment]
I'm fairly sure that Monika asked for my location; I remember being frustrated that my city wasn't on the awc_citylookup.txt, and dealing with it.

Auto_Hair_Change only chooses 'warm/cozy' outfits consistently based off of Auto_Atmos_Change, correct?  If Auto_Atmos_Change wasn't working, Auto_Hair_Change would pick random outfits?  Monika has been consistently wearing warm clothes the past week, rotating through "Sweater (Shoulderless)",  "Hoodie (Green"), and "Jacket (Brown)."  In the weeks before, she was wearing "Sundress (White)" pretty consistently, which to me implies that Auto_Atmos_Change was able to grab the local temperature.

Just now, I repeated the conversation where Monika asks for my location, and afterwards when I toggle the "Auto Sun Times" it no longer return an error, and the Sun Up and down times seem to have updated.  I have to wonder how Auto_Atmos_Change was working before without my location, but with having Monika ask me for my location again, the issue seems to have been resolved.  I've attached my persistent, in case it can give any help in understanding what was going on.

[persistent.zip](https://github.com/multimokia/MAS-Submod-Auto-Atmos-Change/files/5631722/persistent.zip)

**Comment by multimokia** [comment]
Yeah it's odd, I saw a `shown_count` on the topic she asked you, yet somehow your location wasn't stored (by the way, if you could please add the missing location to the file by opening a pull request, that'd be appreciated so nobody else from your area would get excluded like that too)

As for why she'd change into warm clothes, Auto Outfit Change also has a more simplified version of the temperature check, but it's based on time of year and the season.

For example, in the northern hemisphere, she'll use sweaters and warm clothes one month before Winter begins and one month after Winter ends. This is likely why it was working for you.

As for why Auto Atmos Change was working otherwise, no idea. But I'll definitely look into the persistent, thank you.
